clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -31,7 +31,7 @@
     public static final int DEFAULT_THROTTLE_TIME = 0;
 
     protected Send toSend(String destination, ResponseHeader header, short apiVersion) {
-        return new NetworkSend(destination, serialize(apiVersion, header));
+        return new StructSend(destination, header.toStruct(), toStruct(apiVersion));
```

Comment:
With this change, `AbstractResponse#serialize(short, ResponseHeader)` method is used only by unit tests, but not in actual code path -- should we update the relevant unit tests that use this function?